app/controllers/Api2.scala

```diff
@@ -77,13 +75,7 @@ class Api2 (override val stores: DataStores, conf: Configuration, val authAction
 
   def putMediaAtom(id: String) = APIHMACAuthAction { implicit req =>
     parse(req) { atom: MediaAtom =>
-      val thriftAtom = atom.asThrift
-      val atomWithExpiryChecked = expiryPoller.updateStatusIfExpired(thriftAtom) match {
```

Comment:
Expiring a newly created atom is still possible and is now covered by a special case when publishing (below)